src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -132,11 +133,25 @@ public static partial class JsonSerializer
                         clearMax = bytesInBuffer;
                     }
 
+                    int start = 0;
+                    if (loopCount == 0)
+                    {
+                        // Handle the UTF-8 BOM if present
+                        int utf8BomLength = JsonConstants.Utf8Bom.Length;
+                        if (buffer.Length >= utf8BomLength && JsonConstants.Utf8Bom.SequenceEqual(buffer.AsSpan(0, utf8BomLength).Slice(0, utf8BomLength)))
+                        {
+                            start += utf8BomLength;
+                            bytesInBuffer -= utf8BomLength;
+                        }
+                    }
+
+                    loopCount++;
```

Comment:
I don't know this code, but if the only purpose of loopCount is to make sure you only check for the BOM on the first line,  why not make it a bool like 'firstLine' and then you only need to set it within the "if" block and not increment a counter on every loop.

Reply:
Yep, agreed. Instead of an int counter, it can be a bool `firstIteration`.